### PR TITLE
[#21035] Allow users to independently enable Waku telemetry and in-ap…

### LINKED ISSUE #21035: Allow users to independently enable Waku telemetry and in-app analytics
# Feature Issue

### User Stories

1. As a new Status user, I want to make an informed decision about whether I want to help improve Status by sharing usage data, so I'll clearly understand what data will be shared from all profiles and what data will be shared from the current one.

2.  As a more experienced user, I want to have explicit control over the sharing status of data from all profiles and the current profile, so I can better manage the data that might expose my identity.

### Description
Onbording:
- Added Waku telemetry options.
- Created separate sections for "All profiles" and "Current profile" data items.
- Changed the CTA to "Help us improve Status" to address legal concerns. (This avoids phrasing around agreeing to share usage data, as some data will be shared regardless of the decision. The focus is on offering help, not agreeing to data sharing.)

Settings:
- Added Waku telemetry options.
- Created separate sections for "App interactions (Shared from all profiles)" and "Network behavior (Shared from the current profile)" items.
- Updated the bottom copy to address legal concerns.

### Acceptance Criteria

Onboarding:
Design matches (same drawer in 3 places): 
- Create profile https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=13726-23136&t=fASRgmUgi4BQtcGb-1
- Restore profile https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=13726-29112&t=fASRgmUgi4BQtcGb-1
- Update app https://www.figma.com/design/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=14097-4045&t=fASRgmUgi4BQtcGb-1

Settings:
- Design matches: https://www.figma.com/design/JlpPhJp0SMnEyBQy4nOZHo/Settings-for-Mobile?node-id=8477-8868&t=c1MBMhpUvDR9ZthO-1
- Users can toggle both flags independently.

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #21035

### CODE REVIEW

**Comment by ilmotta** [comment]
@xAlisher, correct me if I'm wrong, but I think you are still going to make a few adjustments based on the latest feedback from legal and also to simplify the flow for onboarding as you and I discussed (make it simpler than the experience from within Settings).

If that's the case, then @flexsurfer, it will be preferable to put this PR on hold and come back to it when we have the final decision and designs. We are sorry for coming back & forth on these points, but this time around we want to nail everything and not change it so soon.

**Comment by flexsurfer** [comment]
thanks @ilmotta i guess now it's ready

File: src/status_im/contexts/settings/privacy_and_security/share_usage/view.cljs
Comment (ilmotta) [inline src/status_im/contexts/settings/privacy_and_security/share_usage/view.cljs]: Is it necessary to pass an empty map to `container-style`?

File: src/status_im/contexts/settings/privacy_and_security/share_usage/view.cljs
Comment (ilmotta) [inline src/status_im/contexts/settings/privacy_and_security/share_usage/view.cljs]: The empty style map can be removed, and possibly the entire wrapping `rn/view`?

File: src/status_im/contexts/settings/privacy_and_security/share_usage/view.cljs
Comment (ilmotta) [inline src/status_im/contexts/settings/privacy_and_security/share_usage/view.cljs]: Map style is not inside `{:style ...}`

**Review by ilmotta — APPROVED** [review]
LGTM, thank you for picking this up @flexsurfer 👍🏼 

To make sure we are all happy with the solution, I'm going to add @xAlisher as a reviewer too. @xAlisher, since you worked closely with the designs, would it be possible for you to verify the PR build matches all expectations? Which Figma is now the source of truth?

For the UX I have only two comments, but which are not blockers:

- In the onboarding flow, the privacy bottom sheet can contain more text than fits in smaller devices. This will potentially prevent the user from pressing on "Share usage data" or to read the full Privacy Policy because this bottom sheet is not handling scroll. This is likely not a problem in practice because the majority of our users use larger phones according to app stores.

- I mistakenly accepted to "Share usage data" button, but I actually wanted to open the full Privacy Policy bottom sheet. The cause is because the text "Privacy Policy" is very tiny even in an iPhone 14 Max and too close to the button. My fingers are small, for many users it could be significantly more difficult to open the Privacy Policy without mistakenly accepting to share usage data.

**Comment by mariia-skrypnyk** [comment]
Hey @flexsurfer !

Thanks for your PR.
It's clear for me that there are UI parts that I can test here.
But what about "All profiles" and "Current profile" data types?
How can I test it from my perspective?

**Comment by flexsurfer** [comment]
> But what about "All profiles" and "Current profile" data types?

thanks @mariia-skrypnyk honestly I don't know if it's possible to test, actually there is nothing new, both were already implemented, lets say this PR is only about UI

**Comment by flexsurfer** [comment]
@mariia-skrypnyk Actually you could rely on toggles in the settings

**Comment by mariia-skrypnyk** [comment]
Thanks @flexsurfer !

**Comment by mariia-skrypnyk** [comment]
Hey @flexsurfer !

Here is an issue I've found on both platforms:

**ISSUE 1: Help us modal appears right after logout on switch profile screen**

**Steps:**

1. Install app
2. Create user
3. Do a Logout

**Actual result:**

We see `Help us` modal after a logout

https://github.com/user-attachments/assets/abc20886-06a4-43bf-a0ae-e9dca3b74298


**Expected result:**

No `Help us` modal after a logout.
It appears after we go and create another user.

**Comment by xAlisher** [comment]
Thank you for your work! I found just a couple of small inconsistencies.

For onboarding, the button should read "Help us improve Status."

image

image

Settings: The block style is incorrect, and the bottom copy alignment is off.

image

image

**Comment by flexsurfer** [comment]
hey @xAlisher somehow you used not the latest build, could you take a look at the latest build thx

**Comment by flexsurfer** [comment]
@mariia-skrypnyk oops, pushed dev flag by mistake, should be fixed now

**Comment by mariia-skrypnyk** [comment]
Thanks @flexsurfer !

ISSUE 1 fixed ✅

I started conversation [here] (https://discord.com/channels/1210237582470807632/1217173176086302821/1283734510726414430) regarding "Help to improve" triggering in case we create another Profile. Will wait for a decision.

I've noticed tiny UI difference in Settings modal:

**ISSUE 2: no lockers icons on the modal in "we won't receive" section**

Design vs Implementation

Screenshot 2024-09-12 at 12 45 23
Screenshot 2024-09-12 at 12 45 29

So, for now we only have ISSUE 2 and opened question to discuss.
Thanks.

**Comment by mariia-skrypnyk** [comment]
@flexsurfer 

UPD: 

> we can proactively ask for both types of data only once for the first profile.

https://discord.com/channels/1210237582470807632/1217173176086302821/1283797108490633257

So for now we have only issue with 🔓 icon.

**Comment by flexsurfer** [comment]
thanks @mariia-skrypnyk fixed

**Comment by mariia-skrypnyk** [comment]
Thanks @flexsurfer for adding scroll.
PR can be merged. 
Great job!

**Comment by yevh-berdnyk** [comment]
Hi @flexsurfer, @mariia-skrypnyk, is it expected that we have this modal twice when recovering an account and it's not the first account? Please watch the video here from 4:24 https://app.eu-central-1.saucelabs.com/tests/d089b2d30f3f4855b7761fe7696e4c74?auth=6d07468ca5b76de8b32f8890b89b4d0f#88

**Comment by mariia-skrypnyk** [comment]
Hi @yevh-berdnyk @flexsurfer  !

Sorry, missed to check this tricky scenario:

1. Create a user (choose `Not now` on `Help to improve `modal)
2. Logout
3. Close app
4. Open app (modal appears)
5. Create another profile by any type (recovering or just create)


https://github.com/user-attachments/assets/0eb713b8-c1a9-4583-8ce5-cab78c7c019a

Modal in this case will appear every time you open the app and also appears when you try to create user.
According to the discussion [here](https://discord.com/channels/1210237582470807632/1217173176086302821/1283794833101557874)  we decided to show modal only once when we create the first profile. If user chooses `Not now` he can anytime go to settings and turn on settings. No need to show modal once more.

**Comment by flexsurfer** [comment]
@mariia-skrypnyk thank you, logout should be fixed, not sure about the app reopening

**Comment by mariia-skrypnyk** [comment]
@flexsurfer thanks

Checked on Android on the latest buils and still see incorrect behaviour:

on relogin and on logout too


https://github.com/user-attachments/assets/46cf33c8-7bb1-4da0-9fdf-51797abac318


https://github.com/user-attachments/assets/174cf753-02e6-4b99-b4df-6303844a1fd2

**Comment by flexsurfer** [comment]
thanks @mariia-skrypnyk should be fixed now

**Comment by mariia-skrypnyk** [comment]
Hi @flexsurfer !

Thanks for your final fix🔧 and now modal works correct.
Failed e2e are not related and you can merge the PR. 🙌

**Pull request merged**
